### PR TITLE
(maint) Update pl-cmake to 3.12.4 for Solaris

### PR DESCRIPTION
**Pull request by caseywilliams**
Most platforms use system cmake in puppet-agent, but Solaris is one of the few families that still uses a cmake from pl-build-tools.

pl-cmake at 3.2.3 is too old to use `add_compile_definitions`, which we'd like to use for leatherman, so this updates the pl-cmake package for Solaris only to 3.12.4. This built for me successfully here: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/vanagon-generic-builder/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/304/. I'm testing this with leatherman now, and I'll unblock this once I've confirmed it's working for us.

### CODE REVIEW

File: configs/components/cmake.rb [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+    pkg.url "https://cmake.org/files/v3.2/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
+    pkg.md5sum "d51c92bf66b1e9d4fe2b7aaedd51377c"
+  end
 
   pkg.mirror "#{settings[:buildsources_url]}/#{pkg.get_name}-#{pkg.get_version}.tar.gz"
 
   if platform.name =~ /sles-12/
     pkg.apply_patch 'resources/patches/cmake/avoid-select-sles-12.patch'
-  elsif platform.is_solaris?
-    pkg.apply_patch 'resources/patches/cmake/use-g++-as-linker-solaris.patch'
```
Comment (speedofdark) [inline configs/components/cmake.rb]: Do we have any guidelines around deleting patch files that have (potentially) become obsolete?

Comment (caseywilliams) [reply configs/components/cmake.rb]: Not that I know of, but this patch only applies to the older cmake package on solaris, which (with these changes) no longer exists, so I don't think it makes sense to keep it.

Comment (caseywilliams) [reply configs/components/cmake.rb]: (I've removed that patch file now, too)

**Comment by caseywilliams** [comment]
This does seem to allow for building the master branch of leatherman -- I'm removing the "do not merge" label.

**Comment by branan** [comment]
I'm happy to land this, but I'm not sure what the path to getting updated build-tools into repos is these days. I'm unsure if @McdonaldSeanp or @ScottGarman still have the right permissions? If not we might need somebody like @mwaggett to do it for us?

**Pull request merged**